### PR TITLE
Update to eslint 4.1.0

### PR DESCRIPTION
**Pull request by Avaq**
* Configure rules that were added since 3.5.0
* Separate module config from es6 config

@piotr-gawlowski How do these configurations look to you?
@markelog Do you see anything that conflicts with ESLints intended use?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -53,6 +53,10 @@ module.exports = {
     'no-label-var': [2],
     'no-shadow-restricted-names': [2],
     'no-use-before-define': [2],
+    'func-name-matching': [2],
+    'no-useless-return': [2],
+    'nonblock-statement-body-position': [2, 'beside'],
```
Comment (piotr-gawlowski) [inline index.js]: If this qualifies - would be nice to have it.
```
if (!foo) return;
baz();
```

File: index.js [hunk 2]
```diff
@@ -53,6 +53,10 @@ module.exports = {
...
+    'for-direction': [2],
```
Comment (piotr-gawlowski) [inline index.js]: So looping down is disallowed? I almost never use this anymore but used too..
```javascript
var arr = [1,2,3];
for (var i = arr.length; i > 0; i--) {
}
```

File: index.js [hunk 3]
```diff
@@ -97,6 +101,19 @@ module.exports = {
     'computed-property-spacing': [2, 'never'],
     'key-spacing': [2, {beforeColon: false, afterColon: true, mode: 'minimum'}],
     'brace-style': [2, '1tbs', {allowSingleLine: true}]
+    'switch-colon-spacing': [2, {after: true, before: false}],
+    'array-bracket-newline': [2, {multiline: true}],
+    'array-element-newline': [2, {multiline: true}],
+    'capitalized-comments': [2, 'always', {ignoreConsecutiveComments: true}],
```
Comment (piotr-gawlowski) [inline index.js]: I just check some examples of my comments. Why we put so many restrictions on them?
I think I never started my comment with capital - especially single line comments

File: index.js (hunk 1 above)
Comment (Avaq) [inline index.js]: Yeah. This rule allows:

```js
if (condition) expression
```

But not:

```js
if (condition)
  expression
```

File: index.js (hunk 2 above)
Comment (Avaq) [inline index.js]: Looping down with a "less than" condition is disallowed.

Comment (piotr-gawlowski) [inline index.js]: :+1:

**Pull request merged**